flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSource.java

```diff
@@ -486,6 +493,12 @@ protected KafkaSource<RowData> createKafkaSource(
                 break;
         }
 
+        if (recordEvaluator != null) {
+            kafkaSourceBuilder
+                    .setBounded(new NoStoppingOffsetsInitializer())
```

Comment:
Why do we need this change? I suppose the boundedness of kafka source should be independent of whether `recordEvaluator` is specified.